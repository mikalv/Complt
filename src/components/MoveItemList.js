import React from 'react';
import List from 'react-md/lib/Lists/List';
import ListItem from 'react-md/lib/Lists/ListItem';
import PropTypes from '../PropTypes';

export function renderListItem(item, onChooseItem, itemToMove, key) {
  if (item === null || !item.isProject || item._id === itemToMove) return null;
  return (
    <ListItem
      isOpen
      key={key}
      expanderIconChildren=""
      primaryText={item.name || 'Projects'}
      onClick={() => onChooseItem(item._id)}
      nestedItems={item.children.map((itemInMap, i) =>
        renderListItem(itemInMap, onChooseItem, itemToMove, i))}
    />
  );
}

const MoveItemList = props => (
  <div style={{ overflow: 'scroll', width: '500px' }}>
    <List style={{ overflow: 'scroll', width: '500px' }}>
      {renderListItem(props.itemTree, props.onChooseItem, props.itemToMove, 1)}
    </List>
  </div>

);

MoveItemList.propTypes = {
  itemTree: React.PropTypes.arrayOf(PropTypes.item),
  itemToMove: React.PropTypes.string,
  onChooseItem: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
};

export default MoveItemList;
