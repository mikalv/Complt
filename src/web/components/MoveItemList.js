import React from 'react';
import List from 'react-md/lib/Lists/List';
import ListItem from 'react-md/lib/Lists/ListItem';
import PropTypes from '../../common/PropTypes';

export function renderListItem(item, onChooseItem, itemToMove, key) {
  if (item === null || !item.isProject || item._id === itemToMove) return null;
  return (
    <ListItem
      isOpen
      key={key}
      expanderIconChildren=""
      primaryText={item.name || 'Projects'}
      onClick={() => onChooseItem(item._id)}
      nestedItems={item.children.map((itemInMap, i) =>
        renderListItem(itemInMap, onChooseItem, itemToMove, i))}
    />
  );
}

const MoveItemList = props => (
  <div>
    <List>
      {renderListItem(props.itemTree, props.onChooseItem, props.itemToMove, 1)}
    </List>
  </div>

);

MoveItemList.propTypes = {
  itemTree: PropTypes.item,
  itemToMove: React.PropTypes.string,
  onChooseItem: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
};

export default MoveItemList;
