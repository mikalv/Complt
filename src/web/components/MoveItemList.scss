button.mdc-list-item {
  background: transparent;
  color: inherit;
  border: 0;
  font-size: inherit;
  width: 100%;
}