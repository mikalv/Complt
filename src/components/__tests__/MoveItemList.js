import React from 'react';
import { mount } from 'enzyme';
import MoveItemList from '../MoveItemList';


const tree = {
  isProject: true,
  _id: 'root',
  children: [
    { isProject: true, children: [], name: 'Some Project', _id: 'item1' },
    { isProject: true, children: [], name: 'Some Other Project', _id: 'item2' },
    null,
    { isProject: false, isCompleted: false, tags: [], name: 'Some Task', _id: 'item3' },
  ] };

describe('UpdateItemDialog component', () => {
  it('renders correctly', () => {
    const component = mount(<MoveItemList itemTree={tree} itemToMove="item1" />);
    expect(component).toMatchSnapshot();
  });
  it('calls onChooseItem correctly', () => {
    const onChooseItem = jest.fn();
    const component = mount(<MoveItemList itemTree={tree} onChooseItem={onChooseItem} />);
    component.find('ListItemText').first().simulate('click');
    expect(onChooseItem).toBeCalledWith('root');
  });
});
