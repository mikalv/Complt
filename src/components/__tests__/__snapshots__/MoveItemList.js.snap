exports[`UpdateItemDialog component renders correctly 1`] = `
<Component
  itemToMove="item1"
  itemTree={
    Object {
      "_id": "root",
      "children": Array [
        Object {
          "_id": "item1",
          "children": Array [],
          "isProject": true,
          "name": "Some Project",
        },
        Object {
          "_id": "item2",
          "children": Array [],
          "isProject": true,
          "name": "Some Other Project",
        },
        null,
        Object {
          "_id": "item3",
          "isCompleted": false,
          "isProject": false,
          "name": "Some Task",
          "tags": Array [],
        },
      ],
      "isProject": true,
    }
  }>
  <div>
    <List>
      <ul
        className="md-list">
        <ListItem
          activeClassName="md-text--theme-primary"
          component="div"
          expanderIconChildren=""
          isOpen={true}
          nestedItems={
            Array [
              null,
              <ListItem
                activeClassName="md-text--theme-primary"
                component="div"
                expanderIconChildren=""
                isOpen={true}
                nestedItems={Array []}
                onClick={[Function]}
                primaryText="Some Other Project" />,
              null,
              null,
            ]
          }
          onClick={[Function]}
          primaryText="Projects">
          <li
            className="md-list-item md-list-item--nested-container">
            <AccessibleFakeButtonInked
              __SUPER_SECRET_REF__={[Function]}
              aria-expanded={true}
              className="md-list-tile md-text"
              component="div"
              inkTransitionEnterTimeout={450}
              inkTransitionLeaveTimeout={300}
              inkTransitionOverlap={150}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}>
              <AccessibleFakeButton
                aria-expanded={true}
                className="md-list-tile md-text"
                component="div"
                ink={
                  <InkContainer
                    className={undefined}
                    disabledInteractions={undefined}
                    inkClassName={undefined}
                    inkStyle={undefined}
                    style={undefined}
                    transitionEnterTimeout={450}
                    transitionLeaveTimeout={300}
                    transitionOverlap={150}
                    waitForInkTransition={undefined} />
                }
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}>
                <div
                  aria-expanded={true}
                  aria-pressed={false}
                  className="md-fake-btn md-pointer--hover md-list-tile md-text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}>
                  <InkContainer
                    transitionEnterTimeout={450}
                    transitionLeaveTimeout={300}
                    transitionOverlap={150}>
                    <ReactTransitionGroup
                      childFactory={[Function]}
                      className="md-ink-container"
                      component="div">
                      <div
                        className="md-ink-container" />
                    </ReactTransitionGroup>
                  </InkContainer>
                  <TileAddon
                    activeClassName="md-text--theme-primary" />
                  <ListItemText
                    activeClassName="md-text--theme-primary"
                    className=""
                    primaryText="Projects">
                    <div
                      className="md-tile-content">
                      <div
                        className="md-tile-text--primary md-text">
                        Projects
                      </div>
                    </div>
                  </ListItemText>
                  <TileAddon
                    avatar={null}
                    icon={
                      <Collapser
                        flipped={true}
                        iconClassName={undefined} />
                    }>
                    <div
                      className="md-tile-addon md-tile-addon--icon">
                      <Collapser
                        flipped={true}>
                        <FontIcon
                          className="md-collapser md-collapser--flipped"
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons md-collapser md-collapser--flipped" />
                        </FontIcon>
                      </Collapser>
                    </div>
                  </TileAddon>
                </div>
              </AccessibleFakeButton>
            </AccessibleFakeButtonInked>
            <Collapse
              collapsed={false}
              springConfig={
                Object {
                  "precision": 0.5,
                }
              }>
              <Motion
                defaultStyle={
                  Object {
                    "height": undefined,
                    "paddingBottom": NaN,
                    "paddingTop": NaN,
                  }
                }
                style={
                  Object {
                    "height": NaN,
                    "paddingBottom": NaN,
                    "paddingTop": NaN,
                  }
                }>
                <List>
                  <ul
                    className="md-list md-list--nested-1">
                    <ListItem
                      activeClassName="md-text--theme-primary"
                      component="div"
                      expanderIconChildren=""
                      isOpen={true}
                      nestedItems={Array []}
                      onClick={[Function]}
                      primaryText="Some Other Project">
                      <li
                        className="md-list-item md-list-item--nested-container">
                        <AccessibleFakeButtonInked
                          __SUPER_SECRET_REF__={[Function]}
                          aria-expanded={true}
                          className="md-list-tile md-text"
                          component="div"
                          inkTransitionEnterTimeout={450}
                          inkTransitionLeaveTimeout={300}
                          inkTransitionOverlap={150}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}>
                          <AccessibleFakeButton
                            aria-expanded={true}
                            className="md-list-tile md-text"
                            component="div"
                            ink={
                              <InkContainer
                                className={undefined}
                                disabledInteractions={undefined}
                                inkClassName={undefined}
                                inkStyle={undefined}
                                style={undefined}
                                transitionEnterTimeout={450}
                                transitionLeaveTimeout={300}
                                transitionOverlap={150}
                                waitForInkTransition={undefined} />
                            }
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}>
                            <div
                              aria-expanded={true}
                              aria-pressed={false}
                              className="md-fake-btn md-pointer--hover md-list-tile md-text"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseLeave={[Function]}
                              onMouseOver={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}>
                              <InkContainer
                                transitionEnterTimeout={450}
                                transitionLeaveTimeout={300}
                                transitionOverlap={150}>
                                <ReactTransitionGroup
                                  childFactory={[Function]}
                                  className="md-ink-container"
                                  component="div">
                                  <div
                                    className="md-ink-container" />
                                </ReactTransitionGroup>
                              </InkContainer>
                              <TileAddon
                                activeClassName="md-text--theme-primary" />
                              <ListItemText
                                activeClassName="md-text--theme-primary"
                                className=""
                                primaryText="Some Other Project">
                                <div
                                  className="md-tile-content">
                                  <div
                                    className="md-tile-text--primary md-text">
                                    Some Other Project
                                  </div>
                                </div>
                              </ListItemText>
                              <TileAddon
                                avatar={null}
                                icon={
                                  <Collapser
                                    flipped={true}
                                    iconClassName={undefined} />
                                }>
                                <div
                                  className="md-tile-addon md-tile-addon--icon">
                                  <Collapser
                                    flipped={true}>
                                    <FontIcon
                                      className="md-collapser md-collapser--flipped"
                                      iconClassName="material-icons">
                                      <i
                                        className="md-icon material-icons md-collapser md-collapser--flipped" />
                                    </FontIcon>
                                  </Collapser>
                                </div>
                              </TileAddon>
                            </div>
                          </AccessibleFakeButton>
                        </AccessibleFakeButtonInked>
                        <Collapse
                          collapsed={false}
                          springConfig={
                            Object {
                              "precision": 0.5,
                            }
                          }>
                          <Motion
                            defaultStyle={
                              Object {
                                "height": undefined,
                                "paddingBottom": NaN,
                                "paddingTop": NaN,
                              }
                            }
                            style={
                              Object {
                                "height": NaN,
                                "paddingBottom": NaN,
                                "paddingTop": NaN,
                              }
                            }>
                            <List>
                              <ul
                                className="md-list md-list--nested-2" />
                            </List>
                          </Motion>
                        </Collapse>
                      </li>
                    </ListItem>
                  </ul>
                </List>
              </Motion>
            </Collapse>
          </li>
        </ListItem>
      </ul>
    </List>
  </div>
</Component>
`;
